src/helpers/utils.ts

```diff
@@ -61,8 +61,7 @@ export async function getProposal(id) {
     if (result.errors) {
       console.error(`[events] Errors in subgraph request for proposal id: ${id}`);
     }
-    proposal = result.proposal || null;
-    return proposal;
+    return result.proposal?.flagged ? null : result.proposal || null;
```

Comment:
More confusing 🙈  how about 
```suggestion
    return result.proposal && !result.proposal.flagged ? result.proposal : null;
```